url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/Emit.java

```diff
@@ -0,0 +1,416 @@
+package backend;
+
+import backend.InstructionSet.*;
+import backend.instructions.InsInterface2;
+import backend.instructions.InsInterface1;
+import backend.instructions.InsInterface3;
+import backend.instructions.InsInterface4;
+import frontend.FrontEnd;
+import java.awt.Color;
+import java.util.Hashtable;
+import java.util.StringTokenizer;
+import javax.swing.JOptionPane;
+
+/**
+ *
+ * @author nikhil
+ */
+public class Emit
+{
+
+    public static Hashtable ht;
+    public static String ins;
+
+    public Emit()
+      {
+        ht = new Hashtable();
+      }
+
+    public void createHash()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks such as on lines 33-38, lines 39-44, lines 46-51, lines 53-58, lines 59-64 into separate methods.